src/containers/upload-page/store/sagas.js

```diff
@@ -1031,32 +1033,33 @@ function* uploadTracksAsync(action) {
       )
       return
     }
-    const newEndpoint = selectedServices.join(',')
-
-    // Try to upgrade to creator, early return if failure
-    try {
-      console.debug(`Attempting to upgrade user ${user.user_id} to creator`)
-      yield call(AudiusBackend.upgradeToCreator, newEndpoint)
-    } catch (err) {
-      console.error(`Upgrade to creator failed with error: ${err}`)
-      yield put(uploadActions.uploadTrackFailed())
-      yield put(uploadActions.upgradeToCreatorError(err))
-      return
-    }
+    newEndpoint = selectedServices.join(',')
+  }
 
-    yield put(
-      cacheActions.update(Kind.USERS, [
-        {
-          id: user.user_id,
-          metadata: {
-            creator_node_endpoint: newEndpoint,
-            is_creator: true
-          }
-        }
-      ])
-    )
+  // Try to upgrade to creator, early return if failure
+  let metadata = { is_creator: true }
+  try {
+    // If new endpoint was selected via the selection logic above, update
+    // the user's creator_node_endpoint and is_creator flag
+    console.debug(`Attempting to upgrade user ${user.user_id} to creator`)
+    yield call(AudiusBackend.upgradeToCreator, newEndpoint)
```

Comment:
is it worth adding a check to error if `newEndpoint` is not set at this point?

Reply:
newEndpoint should always be set to the actual replica set assigned to the user. updated so that newEndpoint always has a non-empty value